### PR TITLE
Cleanup/docs updates

### PR DESCRIPTION
**Pull request by guidezpl**
*  Updates diagram_capture example to Android embedding v2
*  Improve/fix documentation
* Create asset output subdirectory if it doesn't already exist (e.g. `assets/cupertino`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,47 +20,47 @@ material library would go in the `assets/material/` directory and be at
...
 
-See the [`generate.dart`](./bin/generate.dart) script. It will
-regenerate almost all of existing assets using the Flutter version you
-have installed.
+The [`generate.dart`](./bin/generate.dart) script regenerates almost all of existing assets
+using the Flutter version you have installed. A small wrapper [`bin/generate.sh`](./bin/generate.sh)
+is provided as a convenience.
 
-Typically, you plug in an Android phone and run:
+You may with to limit image generation to certain categories and/or names. For example:
```
Comment (perclasson) [inline README.md:33]: nit: To limit image generation to certain categories and/or names, run:

File: README.md [hunk 2]
```diff
@@ -20,47 +20,47 @@ material library would go in the `assets/material/` directory and be at
...
+```sh
+# Filter by category
+bin/generate.sh -c cupertino,material
+# Filter by name
+bin/generate.sh -n basic_material_app,blend_mode
+```
 
+You may also specify a different device to use for generating diagrams. For example:
```
Comment (perclasson) [inline README.md:41]: nit: `To specify a different device to use for generating diagrams, run:`

File: README.md [hunk 3]
```diff
@@ -20,47 +20,47 @@ material library would go in the `assets/material/` directory and be at
...
 
-The generator currently only supports running on an Android runtime. An Android
-device or emulator must be running before invoking the `generate.dart` script.
-
-The Android `adb` command, the `flutter` command, and the `dart`
-command need to both be available and in a directory in the `PATH`
-environment variable. For `adb`, be sure it is the same one that is
-running as a server (which is often started by your IDE, so use the
-same `adb` the IDE is running).
+`flutter`, `dart` (and when using an Android device, `adb') commands need to be available
```
Comment (perclasson) [inline README.md:59]: adb here is not correctly quoted

File: utils/diagram_generator/lib/main.dart [hunk 4]
```diff
@@ -36,8 +36,9 @@ Future<Directory> prepareOutputDirectory(String outputDir) async {
 
 Future<void> main() async {
   DiagramFlutterBinding.ensureInitialized();
-  final List<String> arguments = window.defaultRouteName.length > 5
-      ? Uri.decodeComponent(window.defaultRouteName.substring(5)).split(' ')
+  const String stringArgs = String.fromEnvironment('args');
+  final List<String> arguments =  stringArgs != ''
```
Comment (perclasson) [inline utils/diagram_generator/lib/main.dart:40]: nit: double spaces here after =

**Review by perclasson — APPROVED** [review]
LGTM with some nits

File: bin/generate.dart [hunk 5]
```diff
@@ -337,9 +340,8 @@ Future<void> main(List<String> arguments) async {
   final String deviceId = flags['device-id'] as String;
   bool keepTemporaryDirectory = flags['keep-tmp'] as bool;
   String tmpDirFlag = (flags['tmpdir'] as String) ?? '';
-  if (tmpDirFlag.isEmpty && deviceId == 'linux') {
-    // On linux, we can't pass command line arguments to a Flutter app, so we
-    // just use a well-known location for the output.
+  if (tmpDirFlag.isEmpty && !deviceId.startsWith('android')) {
+    // Use use a well-known location for the output.
```
Comment (rami-a) [inline bin/generate.dart:344]:
```suggestion
    // Use a well-known location for the output.
```

**Review by rami-a — APPROVED** [review]
lgtm

**Comment by gspencergoog** [comment]
Thank you so much for thinking about this, @guidezpl but we're actually moving away from generating these using mobile systems. Since the screenshots can all be captured on Linux (working) or macOS (in progress), and we can finally pass command line arguments, there's no need for the complicated data transfer from the mobile systems. And since Flutter does all its own rendering, we can render Cupertino or other widgets on any platform.

Ideally in the future, these will all be generated by CI systems when we modify diagram code.

**Comment by guidezpl** [comment]
Sounds good @gspencergood, I've repurpose this PR to for cleanup/improving docs.

**Comment by gspencergoog** [comment]
@guidezpl The icons from the `cupertino_icons` package should take care of themselves, and we can add some font files to get the right fonts.  I suspect we'll have to set `debugDefaultTargetPlatformOverride` for those diagrams that are Cupertino-based to get the right fonts, behavior, and icons.

I haven't put together a PR yet for the macos work, but I can do that this week.

**Review by gspencergoog — APPROVED** [review]
LGTM, Thanks for the remaining updates in this PR.  They look fine to me.

**Comment by guidezpl** [comment]
I was thinking of app bar back icons and the like. But I think you're right that `debugDefaultTargetPlatformOverride` will handle those differences. Thanks, I'll draft a PR with the code for cupertino screenshots and update them after your PR merges

**Comment by wprater** [comment]
> Thank you so much for thinking about this, @guidezpl but we're actually moving away from generating these using mobile systems. Since the screenshots can all be captured on Linux (working) or macOS (in progress), and we can finally pass command line arguments, there's no need for the complicated data transfer from the mobile systems. And since Flutter does all its own rendering, we can render Cupertino or other widgets on any platform.

@gspencergoog I'd love to hear more about where we can track progress on this?  Im hoping to use a tool like this one to create snapshots of our custom component library when we document it with `dartdoc`.  Ideally we'd be doing this in our pipelines.

**Pull request merged**